YololCompetition/Services/Messages/DBMessages.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.Common;
+using System.Threading.Tasks;
+using Microsoft.Data.Sqlite;
+using YololCompetition.Services.Database;
+using YololCompetition.Services.Cron;
+using YololCompetition.Services.Challenge;
+using YololCompetition.Extensions;
+using Discord.WebSocket;
+using Discord;
+
+namespace YololCompetition.Services.Messages
+{
+    public class DbMessages
+        : IMessages
+    {
+        private readonly IDatabase _database;
+        private readonly ICron _cron;
+        private readonly IChallenges _challenges;
+        private readonly BaseSocketClient _client;
+
+        public enum messageType : uint {Current, Leaderboard}
+        
+        public DbMessages(IDatabase database, ICron cron, IChallenges challenges, BaseSocketClient client)
```

Comment:
You either need to take `DiscordSocketClient` or add `BaseSocketClient` into the DI container in `Program.cs` (at the moment only `DiscordSocketClient` is added iirc, it's probably better to add both)

Reply:
DiscordSocketClient has the methods I need so I don't see why both would be better. I've changed it to take DiscordSocketClient

Reply:
Sorry I wasn't very clear about that. What I mean is in `Program.cs` this code:

```
di.AddSingleton(new DiscordSocketClient(new DiscordSocketConfig {
    AlwaysDownloadUsers = true,
    MessageCacheSize = 0,
}));
```

Binds `DiscordSocketClient` into the DI container. However `BaseSocketClient` is _not_ bound into the container anywhere.

So what I was saying is you can either change your constructor to request `DiscordSocketClient` (since that's available in DI). Or you could change `Program.cs` to bind _both_ `DiscordSocketClient` and `BaseSocketClient` to the same item in DI, and then your constructor can continue requesting `BaseSocketClient`.